client/src/stores/balances.js

```diff
@@ -21,3 +22,52 @@ auth.subscribe(async auth => {
 		console.error('Error while fething balances:', err)
 	}
 })
+
+/**
+ * Atualiza o Balance Locked ao receber uma nova transaction
+ */
+addSocketListener('new_transaction', (currency, transaction) => {
+	transactions.update(tx => {
+		tx.push(transaction)
+		return tx
+	})
+	update(balances => {
+		if (transaction.status === 'confirmed') {
+			balances[currency].available += transaction.amount
+		} else {
+			balances[currency].locked += transaction.amount
+		}
+		return balances
+	})
+})
+
+/**
+ * Após receber a confirmação da transação ele atualiza o Balance Available
+ */
+addSocketListener('update_received_tx', async (currency, txUpdate) => {
+	console.log(txUpdate)
+	if (txUpdate.status === 'confirmed') {
+		let txInfo = transactions.getTxByOpid(txUpdate.opid)
+		if (txInfo) {
+			update(balances => {
+				balances[currency].available += +txInfo.amount
+				balances[currency].locked -= +txInfo.amount
+				return balances
+			})
+		} else {
+			try {
+				/** Pega os dados da transação pelo opid */
+				txInfo = await emit('get_tx_info', txUpdate.opid)
+				console.log(txInfo)
+				/** Usa dados do amount pego no 'txInfo' para atualizar o saldo na tela */
+				update(balances => {
+					balances[currency].available += +txInfo.amount.$numberDecimal
+					balances[currency].locked -= +txInfo.amount.$numberDecimal
```

Comment:
WTF?

Reply:
esse é o retorno do get_tx_info 
`{ 
     amount: {
        $numberDecimal: "2e-7"
     }
}`

Reply:
Mas ele deveria retornar um number, se n tá retornando é um bug

Pq vc não comentou disso?

Reply:
Dessa vez eu esqueci mesmo

Reply:
Quando o bug for consertado eu vou tirar esse segundo update, já que ele esta ai só por conta disso

Reply:
aaaai carai...

Vou corrigir isso daqui a pouco, já pode presumir que vai ser um
number daqui em diante

Reply:
Fixed!